src/riscv.c

```diff
@@ -325,3 +325,19 @@ void state_delete(state_t *s)
     memory_delete(s->mem);
     free(s);
 }
+
+#if RV32_HAS(JIT)
+void profile(block_t *block)
+{
+    printf("%#-8x|", block->pc_start);
```

Comment:
Instead of displaying messages, configure them to be written to files. You can use the `dump_registers` function as a reference for this process. Furthermore, ensure the IR sequence is logged in files for future reference and analysis. This implies the need to develop a standalone tool specifically for analyzing the profiling data.